src/components/views/MyCommunities.jsx

```diff
@@ -44,26 +44,27 @@ class MyCommunities extends Component {
     if (prevProps.currentUser !== this.props.currentUser) {
       // eslint-disable-next-line react/no-did-update-set-state
       this.setState({ isLoading: true });
-      if (this.communitiesObserver) this.communitiesObserver.unsubscribe();
+      CommunityService.unsubscribe();
```

Comment:
You have considered only one component will subscribe in each view. It can not be a safe assumption. Please change it for other similar subscriptions in other files